staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware_test.go

```diff
@@ -222,6 +223,13 @@ func TestServeHTTP(t *testing.T) {
 				"Access-Control-Allow-Methods",
 			},
 		},
+		{
```

Comment:
I haven't looked at the other tests but can we add a test that asserts this isn't followed when not set?

Reply:
The other tests were already implicitly testing this, however I added another test to make this explicit